llvm/unittests/Target/AIE/AIEScheduleDAGMITest.cpp

```diff
@@ -0,0 +1,64 @@
+//===- AIEScheduleDAGMITest.cpp -------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+
+#include "AIEHazardRecognizer.h"
+#include "AIEMachineScheduler.h"
+#include "ScheduleDAGMITestUtils.h"
+
+using namespace llvm;
+
+namespace {
+
+class DummyAIEHazardRecognizer : public AIEHazardRecognizer {
```

Comment:
Could we derive from a more base HazardRecognizer?

Reply:
Not really, `computeAndFinalizeBundles` expects a `AIEHazardRecognizer`